hyper-capi/include/hyper.h

```diff
@@ -0,0 +1,286 @@
+#ifndef _HYPER_H
+#define _HYPER_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <stddef.h>
+
+typedef struct hyper_clientconn hyper_clientconn;
+
+typedef struct hyper_clientconn_options hyper_clientconn_options;
+
+typedef struct hyper_io hyper_io;
+
+typedef struct hyper_request hyper_request;
+
+typedef struct hyper_response hyper_response;
+
+typedef struct hyper_headers hyper_headers;
+
+typedef struct hyper_body hyper_body;
+
+typedef struct hyper_buf hyper_buf;
+
+typedef struct hyper_task hyper_task;
+
+typedef struct hyper_context hyper_context;
+
+typedef struct hyper_waker hyper_waker;
+
+typedef struct hyper_executor hyper_executor;
+
+// A string reference.
+//
+// The data in here typically is not owned by this struct.
+typedef struct hyper_str {
+	const uint8_t *buf;
+	size_t len;
+} hyper_str;
+
+typedef enum {
+    HYPERE_OK,
+    HYPERE_KABOOM,
+} hyper_error;
+
+typedef enum {
+	HYPER_IT_CONTINUE,
+	HYPER_IT_BREAK,
+} hyper_iter_step;
+
+typedef enum {
+	HYPER_POLL_READY,
+	HYPER_POLL_PENDING,
+} hyper_poll;
+
+// HTTP ClientConn
+
+// Starts an HTTP client connection handshake using the provided IO transport
+// and options.
+//
+// Both the `io` and the `options` are consumed in this function call.
+//
+// The returned `hyper_task *` must be polled with an executor until the
+// handshake completes, at which point the value can be taken.
+hyper_task *hyper_clientconn_handshake(hyper_io *io, hyper_clientconn_options *options);
+
+// Send a request on the client connection.
+//
+// Returns a task that needs to be polled until it is ready. When ready, the
+// task yields a `hyper_response *`.
+hyper_task *hyper_clientconn_send(hyper_clientconn *client, hyper_request *request);
+
+// Creates a new set of HTTP clientconn options to be used in a handshake.
+hyper_clientconn_options *hyper_clientconn_options_new();
```

Comment:
In C, you should put a `void` between those parentheses, like `(void)` if there's no argument. `-Wstrict-prototypes` will tell you:
~~~
./include/hyper.h:76:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
   76 | hyper_clientconn_options *hyper_clientconn_options_new();
      | ^~~~~~~~~~~~~~~~~~~~~~~~
./include/hyper.h:92:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
   92 | hyper_io *hyper_io_new();
      | ^~~~~~~~
./include/hyper.h:132:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  132 | hyper_request *hyper_request_new();
      | ^~~~~~~~~~~~~
./include/hyper.h:232:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
  232 | hyper_executor *hyper_executor_new();
      | ^~~~~~~~~~~~~~